tests/config_file_test.py

```diff
@@ -39,128 +39,109 @@ def assert_eq(self, result, expected, error_message):
+            '''
+            self.assertIsInstance(exp, dict, 'expected is not dict')
+            self.assertIsInstance(res, dict, 'result is not dict')
+            
+            self.assertEqual(len(exp.keys()), len(res.keys()), 'expected and result dicts don\'t have the same number of keys')
+    
+            for key in exp.keys():
+                assert_sub_value_eq(exp[key], res[key])
+            
+        try:
+            assert_sub_dict_eq(exp, res)
+        except AssertionError as e:
+            raise AssertionError('%s\n%s' % (err_msg, e.message))
+        
+    def test_load_root_config(self):
+        '''
+        tests ConfigFileLoader.load_root_config by inputing a root configuration
+        file from the specified test file path, and asserts that the resulting
+        processed content has the file references properly updated by comparing
+        it against an expected results file, localized to the test platform
+        '''
+        yml = ConfigFileLoader.load_root_config('tests/test_files/root_config.yml')
+
+        # load expected result
+        with open('tests/test_files/root_config_expected.yml', 'r', 1) as input_file:
+            exp_yml = yaml.load(input_file)
+        
+        # update paths in expected results to absoluate paths on this platform
+        exp_yml['dashboard']['owning'] = os.path.abspath(exp_yml['dashboard']['owning']);
+        exp_yml['dashboard']['accessors']['test-org-1']['enterprise']['priv_key_path'] = os.path.abspath(exp_yml['dashboard']['accessors']['test-org-1']['enterprise']['priv_key_path']);
+        exp_yml['dashboard']['accessors']['test-org-2'] = os.path.abspath(exp_yml['dashboard']['accessors']['test-org-2']);
+        exp_yml['dashboard']['accessor_config_filename_format'] = os.path.abspath(exp_yml['dashboard']['accessor_config_filename_format']);
+        exp_yml['logging']['file_log_directory'] = os.path.abspath(exp_yml['logging']['file_log_directory']);
 
-            # test control keys
-            self.assert_eq(yml['dashboard']['test'], 'value should not change', '/dashboard/test value should not change')
-            self.assert_eq(yml['other']['test-string'], 'test string value should not change', '/other/test-string value should not change')
-            self.assert_eq(yml['other']['test-dict']['test-string-2'], 'this should not change as well', '/other/test-dict/test-string-2 value should not change')
-            self.assert_eq(yml['other']['test-list'][0], 'item-1', '/other/test-list/[0] value should not change')
-            self.assert_eq(yml['other']['test-list'][2]['test-string-3'], 'xyz', '/other/test-list/[2] value should not change')
+        self.assert_dict_eq(yml, exp_yml, 'test root configuration did not match expected configuration')
 
-    @mock.patch('__builtin__.open')
-    @mock.patch('yaml.load')
-    def test_load_root_default_config(self, mock_yaml, mock_open):
+    def test_load_root_default_config(self):
```

Comment:
As per the comment above, then, this function would go away and be replaced with a single test-invoker that ran the test function over a specified list of file groups.  This is similar to the approach taken now to the default identity test stuff.